### ISSUE TITLE
action.yaml: no such file or directory

### ISSUE THREAD

**Issue by oscarmorrison** [nektos/act#246]
my main.yml

```
name: Update Price
on: push

jobs:
  scrape:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [13.x]

    steps:
      - uses: actions/checkout@v2
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1.4.2
        with:
          node-version: ${{ matrix.node-version }}
      - name: npm install, build, & run
        run: |
          npm i -g yarn
          yarn install
      - name: Run price check
        uses: ianwalter/puppeteer@cbdd5c50c8d6b6275cdf46e4ad2b3f7ee61211ce
        with:
            args: yarn run update-price
      - name: Deploy to gh-pages
        uses: peaceiris/actions-gh-pages@v3
        with:
            github_token: ${{ secrets.GITHUB_TOKEN }}
            publish_dir: ./data
```

my .actrc
```
-P ubuntu-latest=nektos/act-environments-ubuntu:18.04
```

Getting an error running it

```
[Update Price]   ❌  Failure - Run price check
Error: open [User directory]/.cache/act/ianwalter-puppeteer@cbdd5c50c8d6b6275cdf46e4ad2b3f7ee61211ce/action.yaml: no such file or directory
```

**Comment by tobsenll** [comment]
I have the same problem when I reference a local DockerAction without a action.yaml but only a Dockerfile.
This is working when running on github but not with act locally.

**Comment by stavalfi** [comment]
@tobsenll @oscarmorrison is there any workaround? i have the same issue when using `actions-hub/docker/login@master`

**Comment by jsoref** [comment]
@tobsenll / @stavalfi: could you provide a testcase?

I'm using this, but I have no idea if it's representative:
```
name: Log into Docker
on: push

jobs:
  login:
    runs-on: ubuntu-latest
    steps:
      - name: Trigger docker login
        uses: actions-hub/docker/login@master
```

**Comment by benwinding** [comment]
Okay, so here's how I got around this error in the `using` statement: 

``` yaml 
uses: some/action@master
```

- `some/action` requires the `action.yaml` file to be included in the action you're referencing [more details here](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions). It doesn't work without it.
- `some/action` doesn't work with `@BRANCH_NAME` (even though it works on Github) you need to use one of the following:
  - `action@master`
  - `action@COMMIT_HASH`
  - `action@TAG_NAME`

**Comment by jsoref** [comment]
@benwinding: I suspect the lack of handling for branch name might be handled for nektos/act#62, otherwise, it deserves its own issue.

**Comment by maxandersen** [comment]
I have the same issue in https://github.com/jbangdev/jbang-action I use:

```
 - name: Login to docker hub
        if: success()
        uses: actions-hub/docker/login@master
        env:
          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
```

and it complains about login/action.yaml not found.

tried grokking what @benwinding suggested in https://github.com/nektos/act/issues/246#issuecomment-662237162 but if one remove `actions-hub` nothing work ;)

**Comment by jsoref** [comment]
This issue is almost certainly still valid (I'm not a fan of that bot)

**Comment by jsoref** [comment]
Bad bot

**Comment by ibraheeam** [comment]
Hey

I am also facing the same issue with this [https://github.com/devmasx/brakeman-linter-action](action)

Thanks

**Comment by rickstaa** [comment]
@github-actions I think this is still relevant.

**Comment by jsoref** [comment]
Bad bot

**Comment by catthehacker** [comment]
Fixed by #293
